### PR TITLE
Add update command

### PR DESCRIPTION
**Pull request by NekrozQliphort**
Multiple changes are made:
- Episode Attribute is added
- Update Command(Only for Episode) is added

Currently still missing:
- [x] Missing documentation
- [x] Missing Testcases
- [ ] Changes to UI

### CODE REVIEW

File: src/main/java/seedu/anilist/model/anime/Anime.java [hunk 1]
```diff
@@ -19,21 +19,37 @@
...
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Anime(Name name, Episode episode, Set<Tag> tags) {
+        requireAllNonNull(name, tags, episode);
+        this.name = name;
+        this.episode = episode;
         this.tags.addAll(tags);
     }
```
Comment (clyveycui) [inline src/main/java/seedu/anilist/model/anime/Anime.java:43]: Has this been linked to the addAnimeCommand yet? 
i.e. you can specify the episode while adding an anime.

File: src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        Anime editedAnime = createUpdatedAnime(animeToEdit, episodeDescriptor);
+
+        model.setAnime(animeToEdit, editedAnime);
+        model.updateFilteredAnimeList(PREDICATE_SHOW_ALL_ANIME);
+        return new CommandResult(String.format(MESSAGE_UPDATE_ANIME_EPISODE_SUCCESS, editedAnime));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
```
Comment (auhc99) [inline src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java:68]: change person to anime

File: src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        }
+
+        // state check
+        UpdateEpisodeCommand e = (UpdateEpisodeCommand) other;
+        return index.equals(e.index)
+            && episodeDescriptor.equals(e.episodeDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the person with. Each non-empty field value will replace the
```
Comment (auhc99) [inline src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java:100]: change person to anime

File: src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+     * corresponding field value of the person.
```
Comment (auhc99) [inline src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java:101]: change person to anime

File: src/main/java/seedu/anilist/model/anime/Anime.java [hunk 5]
```diff
@@ -10,7 +10,7 @@
 import seedu.anilist.model.tag.Tag;
 
 /**
- * Represents a Anime in the address book.
+ * Represents an Anime in the address book.
```
Comment (auhc99) [inline src/main/java/seedu/anilist/model/anime/Anime.java:13]: change address book to anime list

**Review by auhc99 — COMMENTED** [review]
need to change person to anime

File: src/main/java/seedu/anilist/model/anime/Anime.java (hunk 1 above)
Comment (NekrozQliphort) [reply src/main/java/seedu/anilist/model/anime/Anime.java:43]: I don't believe so, I feel like that should be on another branch as the current branch is getting too large. Maybe I'll add this issue.

File: src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+        + ": Updates the episode number of the anime identified "
+        + "by the index number used in the displayed anime list. "
+        + "Existing values will be overwritten by the input values.\n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + PREFIX_EPISODE + "EPISODE\n"
+        + "Example: " + COMMAND_WORD + " 1 " + PREFIX_EPISODE + "1";
+
+    public static final String MESSAGE_UPDATE_ANIME_EPISODE_SUCCESS = "Updated Anime: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "An episode number should be provided";
```
[the remark this answers is not in the document]
Comment (NekrozQliphort) [reply src/main/java/seedu/anilist/logic/commands/UpdateEpisodeCommand.java]: Yup I think you're right!

**Review by luodan01 — APPROVED** [review]
lgtm

**Pull request merged**